aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandler.java

```diff
@@ -81,16 +106,55 @@ private GetPolicyResponse getExistingPolicy(AmazonWebServicesClientProxy proxy,
         );
     }
 
+    private void updateNotificationChannels(String pgName, AmazonWebServicesClientProxy proxy, List<Channel> requestedConfiguration) {
+        List<Channel> currentChannels = getExistingNotificationConfiguration(proxy, pgName).notificationConfiguration().channels();
+        updateNotificationChannels(currentChannels, requestedConfiguration, pgName, proxy);
+    }
+
+    private GetNotificationConfigurationResponse getExistingNotificationConfiguration(AmazonWebServicesClientProxy proxy, String profilingGroupName) {
+        return proxy.injectCredentialsAndInvokeV2(GetNotificationConfigurationRequest.builder()
+                        .profilingGroupName(profilingGroupName)
+                        .build(),
+                profilerClient::getNotificationConfiguration);
+    }
+
+    // Generate the change-set and add / delete notification channel based on change-set
+    private void updateNotificationChannels(List<Channel> currentChannels, List<Channel> requestedConfiguration, String pgName, AmazonWebServicesClientProxy proxy) {
+        Map<String, Channel> currentChannelsMap = currentChannels.stream().collect(Collectors.toMap(Channel::uri, val -> val));
```

Comment:
Minor: There's actually a [`Function.identity()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/function/Function.html#identity()) that does exactly `val -> val`.

Reply:
Note: `identity()` is being used on line 124 but there's another use of it here on line 123.